File: backend/app/schemas/workforce.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class AgentSchema(BaseModel):
    id: str
    name: str
    description: str
    status: str
    progress: int
    executionTime: str
    output: Optional[Dict[str, Any]] = None
    logs: List[str] = []

class MissionStepSchema(BaseModel):
    id: str
    agentId: str
    agentName: str
    stage: str
    status: str
    duration: Optional[str] = None
    output: Optional[Dict[str, Any]] = None
    logs: List[str] = []

class MissionStateSchema(BaseModel):
    missionId: str
    fileName: str
    fileType: str
    fileSize: str
    stage: str
    progressPercentage: int
    totalDuration: str
    currentAgent: Optional[AgentSchema] = None
    completedAgents: List[str]
    remainingAgents: List[str]
    steps: List[MissionStepSchema]
    logs: List[Dict[str, Any]] = []

File: frontend/src/types/workforce.ts
export type MissionStage =
  | 'Uploaded'
  | 'Queued'
  | 'Planner'
  | 'Routing'
  | 'Document Analysis'
  | 'Decision'
  | 'Automation'
  | 'Validation'
  | 'Report Generation'
  | 'Completed'
  | 'Failed';

export type AgentId =
  | 'planner'
  | 'router'
  | 'document'
  | 'decision'
  | 'automation'
  | 'validation'
  | 'report';

export type AgentStatus = 'Idle' | 'Processing' | 'Completed' | 'Failed';

export interface WorkforceAgent {
  id: AgentId;
  name: string;
  description: string;
  status: AgentStatus;
  progress: number;
  executionTime: string;
  input?: Record<string, any>;
  output?: Record<string, any>;
  logs: string[];
}

export interface MissionStep {
  id: string;
  agentId: AgentId;
  agentName: string;
  stage: MissionStage;
  status: 'Pending' | 'In_Progress' | 'Completed' | 'Failed';
  startedAt?: string;
  completedAt?: string;
  duration?: string;
  output?: Record<string, any>;
  logs: string[];
}

export interface ExecutionLog {
  id: string;
  missionId: string;
  agentId: AgentId;
  level: 'INFO' | 'WARN' | 'ERROR';
  message: string;
  timestamp: string;
}

export interface WorkforceMissionState {
  missionId: string;
  fileName: string;
  fileType: string;
  fileSize: string;
  stage: MissionStage;
  progressPercentage: number;
  totalDuration: string;
  currentAgent: WorkforceAgent | null;
  completedAgents: AgentId[];
  remainingAgents: AgentId[];
  steps: MissionStep[];
  logs: ExecutionLog[];
}
